core/src/OpenACC/Kokkos_OpenACC.cpp

```diff
@@ -46,11 +46,13 @@ Kokkos::Experimental::OpenACC::OpenACC()
 }
 
 Kokkos::Experimental::OpenACC::OpenACC(int async_arg)
-    : m_space_instance(new Kokkos::Experimental::Impl::OpenACCInternal,
-                       [](Impl::OpenACCInternal* ptr) {
-                         ptr->finalize();
-                         delete ptr;
-                       }) {
+    : m_space_instance(
+          new Kokkos::Experimental::Impl::OpenACCInternal,
+          [](Impl::OpenACCInternal* ptr) {
+            ptr->finalize();
+            atomic_dec(&Impl::OpenACCInternal::m_num_active_user_asyncs);
```

Comment:
I think the logic is flawed.
This constructor is public.  Assume the user does `OpenACC exec(42);`. Now the `m_num_active_user_asyncs` counter has not been incremented but it will be decremented at destruction.